storage.go

```diff
@@ -366,49 +371,105 @@ func (f *file) Move(fromPath string, toPath string) FileResponse {
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("failed to send http request: %w", err)
+	}
+	if hasCustomError {
+		return nil, &errResp
 	}
 
-	body, err := io.ReadAll(res.Body)
+	resp.SignedURL = f.storage.client.BaseURL + resp.SignedURL
+	return &resp, nil
+}
+
+func (f *file) CreateSignedURLForDownload(filePath string, expiresIn int) (*SignedURLForDownloadResponse, error) {
+	reqBody, err := json.Marshal(map[string]interface{}{
+		"expiresIn": expiresIn,
+	})
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("failed to marshal request body: %w", err)
 	}
 
-	var response SignedUrlResponse
-	if err := json.Unmarshal(body, &response); err != nil {
-		panic(err)
+	// Route for generating signed url for download: /object/sign/:bucketId/:objectKey
+	// See https://supabase.github.io/storage
+	reqURL := fmt.Sprintf("%s/%s/object/sign/%s/%s", f.storage.client.BaseURL, StorageEndpoint, f.BucketId, filePath)
+	req, err := http.NewRequest(http.MethodPost, reqURL, bytes.NewBuffer(reqBody))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create http request: %w", err)
 	}
-	response.SignedUrl = f.storage.client.BaseURL + response.SignedUrl
 
-	return response
+	injectAuthorizationHeader(req, f.storage.client.apiKey)
+	req.Header.Set("Content-Type", "application/json")
+
+	var resp SignedURLForDownloadResponse
+	var errResp FileErrorResponse
+	hasCustomError, err := f.storage.client.sendCustomRequest(req, &resp, &errResp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send http request: %w", err)
+	}
+	if hasCustomError {
+		return nil, &errResp
+	}
+
+	resp.SignedURL = f.storage.client.BaseURL + resp.SignedURL
+	return &resp, nil
 }
 
 // GetPublicUrl get a public signed url of a file object
-func (f *file) GetPublicUrl(filePath string) SignedUrlResponse {
-	var response SignedUrlResponse
-	response.SignedUrl = fmt.Sprintf("%s/%s/object/public/%s/%s", f.storage.client.BaseURL, StorageEndpoint, f.BucketId, filePath)
+func (f *file) GetPublicUrl(filePath string) SignedURLForDownloadResponse {
+	var response SignedURLForDownloadResponse
+	response.SignedURL = fmt.Sprintf("%s/%s/object/public/%s/%s", f.storage.client.BaseURL, StorageEndpoint, f.BucketId, filePath)
 	return response
 }
 
-// Remove deletes a file object
-func (f *file) Remove(filePaths []string) FileResponse {
+// Remove deletes a single object from a storage bucket
+func (f *file) Remove(filePath string) error {
+	// Route for deleting object: /object/:bucketId/:objectKey
+	// See https://supabase.github.io/storage
+	reqURL := fmt.Sprintf("%s/%s/object/%s/%s", f.storage.client.BaseURL, StorageEndpoint, f.BucketId, filePath)
+	req, err := http.NewRequest(http.MethodDelete, reqURL, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create http request: %w", err)
+	}
+
+	injectAuthorizationHeader(req, f.storage.client.apiKey)
+
+	var errResp FileErrorResponse
+	hasCustomError, err := f.storage.client.sendCustomRequest(req, nil, &errResp)
```

Comment:
This construct is pretty weird. Probably because of the `sendCustomRequest` method signature (didn't see it). Leave it as it is for now but this should be refactored in the future to handle errors in an idiomatic way

Reply:
totally agree; this should be addressed during the proper refactoring